src/ui/components/shared/SharingModal/SharingModal.tsx

```diff
@@ -123,9 +127,178 @@ function EnvironmentVariablesRow() {
+            setShowPrivacy={setShowPrivacy}
+          />
+        )}
+      </section>
+    </>
+  );
+}
+
+type ModalMode = "sharing" | "download";
+
+function Header({
+  modalMode,
+  setModalMode,
+}: {
+  modalMode: ModalMode;
+  setModalMode: React.Dispatch<React.SetStateAction<ModalMode>>;
+}) {
+  const Tab = ({
+    label,
+    onClick,
+    active,
+    className = "",
+  }: {
+    label: string;
+    onClick: any;
+    active: boolean;
+    className?: string;
+  }) => (
+    <div
+      onClick={onClick}
+      className={
+        (active
+          ? "rounded-xl bg-menuHoverBgcolor px-4 py-1 font-bold text-slate-700 hover:text-slate-800"
+          : "text-slate-500 hover:text-slate-600") + ` ${className} cursor-pointer`
+      }
+    >
+      {label}
+    </div>
+  );
+
+  return (
+    <section className="justify-left flex flex-row items-center space-x-2  px-8 py-4">
+      <Tab
+        label="Sharing"
+        onClick={() => setModalMode("sharing")}
+        active={modalMode == "sharing"}
+      />
+      <Tab
+        label="Downloads"
+        onClick={() => setModalMode("download")}
+        active={modalMode == "download"}
+      />
+    </section>
+  );
+}
+
+function DownloadSection({ recording }: { recording: Recording }) {
+  const token = useToken();
+  const currentTime = useAppSelector(getCurrentTime);
+  const [screen, setScreen] = useState<ScreenShot | null>(null);
+
+  const [downloadState, setDownloadState] = useState<
+    "not-started" | "downloading" | "success" | "error"
+  >("not-started");
+
+  useEffect(() => {
+    getGraphicsAtTime(currentTime).then(res => res.screen && setScreen(res.screen));
+  }, [currentTime]);
+
+  const buttonStates = {
+    "not-started": {
+      label: "Download video",
+      icon: "download",
+    },
+    downloading: {
+      label: "Downloading video",
+      icon: "loop",
+    },
+    success: {
+      label: "Downloaded video",
+      icon: "check",
+    },
+    error: {
+      label: "Failed to download",
+      icon: "error",
+    },
+  };
+
+  const onDownload = () => {
+    console.log(`Download recording ${recording.id}`);
+
+    const xhr = new XMLHttpRequest();
+    xhr.responseType = "blob";
+    xhr.onload = function () {
+      const a = document.createElement("a");
+      a.href = window.URL.createObjectURL(xhr.response);
+      console.log(`Video downloaded ${recording.id}`);
+      setDownloadState("success");
+
+      a.download = "replay.mp4";
+      document.body.appendChild(a);
+      a.click();
+      document.body.removeChild(a);
+    };
+    xhr.onerror = function () {
+      console.log(`Failed to download video ${recording.id}`);
+      setDownloadState("error");
+    };
+    xhr.open("POST", "/api/video");
+    xhr.setRequestHeader("Content-Type", "application/json");
+    xhr.send(JSON.stringify({ recordingId: recording.id, token: token.token }));
+    console.log(`Downloading recording ${recording.id}`);
```

Comment:
Generally I think we should probably remove `console.log`s from code that runs in the browser. I think those are probably only good to leave in when we're development/debugging stuff.